cdap-app-fabric/src/main/java/io/cdap/cdap/internal/accelerator/AcceleratorManager.java

```diff
@@ -0,0 +1,138 @@
+import io.cdap.cdap.spi.metadata.SearchResponse;
+import org.apache.twill.filesystem.LocationFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Class with helpful methods for dynamic accelerator framework
+ */
+public class AcceleratorManager {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AcceleratorManager.class);
+  public static final String ACCELERATOR_TAG = "accelerator:%s";
+  public static final String APPLICATION_TAG = "application:%s";
+  public static final String ACCELERATOR_KEY = "accelerator";
+  private final MetadataAdmin metadataAdmin;
+  private final CConfiguration cConf;
+  private static LocationFactory locationFactory;
+
+  @Inject
+  AcceleratorManager(MetadataAdmin metadataAdmin, CConfiguration cConf) {
+    this.metadataAdmin = metadataAdmin;
+    this.cConf = cConf;
+  }
+
+  /**
+   * Returns the list of applications that are having metadata tagged with the accelerator
+   *
+   * @param namespace
+   * @param accelerator
+   * @return
+   * @throws Exception
+   */
+  public Set<ApplicationId> getAppsForAccelerator(NamespaceId namespace,
+                                                  String accelerator) throws Exception {
+    String acceleratorTag = String.format(ACCELERATOR_TAG, accelerator);
+    SearchRequest searchRequest = SearchRequest.of(acceleratorTag).addNamespace(namespace.getNamespace())
+      .setScope(MetadataScope.SYSTEM).build();
+    SearchResponse searchResponse = metadataAdmin.search(searchRequest);
+    List<MetadataRecord> results = searchResponse.getResults();
+    Set<ApplicationId> applicationIds = results.stream()
+      .map(MetadataRecord::getEntity)
+      .filter(this::isApplicationType)
+      .map(this::getApplicationId)
+      .collect(Collectors.toSet());
+    return applicationIds;
+  }
+
+  /**
+   * Returns boolean indicating whether application is disabled due to a disabled accelerator
+   *
+   * @param namespace
+   * @param applicationName
+   * @return
+   * @throws Exception
+   */
+  public boolean isApplicationDisabled(String namespace, String applicationName) throws Exception {
+    String applicationQuery = String.format(APPLICATION_TAG, applicationName);
+    SearchResponse searchResponse = metadataAdmin
+      .search(SearchRequest.of(applicationQuery).addNamespace(namespace).setScope(MetadataScope.SYSTEM).build());
+    Set<String> acceleratorTagsForApp = searchResponse.getResults().stream()
+      .filter(this::hasAcceleratorTagValue)
+      .map(this::getAcceleratorTagValue)
+      .collect(Collectors.toSet());
+    for (String acceleratorTagForApp : acceleratorTagsForApp) {
+      if (isAcceleratorDisabled(acceleratorTagForApp)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private String getAcceleratorTagValue(MetadataRecord metadataRecord) {
```

Comment:
Seems like the return value can be null? Please annotate with `@Nullable`.

Reply:
Done